detect_secrets/core/scan.py

```diff
@@ -302,24 +311,27 @@ def _get_lines_from_diff(diff: str) -> Generator[Tuple[str, List[Tuple[int, str]
         yield (
             filename,
             [
-                (line.target_line_no, line.value)
+                (
+                    line.target_line_no if line.target_line_no
+                    else line.source_line_no, line.value, line.is_added, line.is_removed,
+                )
                 for chunk in patch_file
                 # target_lines refers to incoming (new) changes
-                for line in chunk.target_lines()
-                if line.is_added
+                for line in (line for line in chunk)
```

Comment:
```suggestion
                for line in [line for line in chunk]
```
in this case a tuple is not so good, because Python creates with each added line a new tuple object